stanley:
  end_position: [170.71, -104.15, 0.0] # 최종 목적지 (시나리오 에디트 모드에서 확인한 값)
  dash_velocity: 100 # km/h, dash case에서의 목표 속도
  dash_initiation_distance:
    200 # 최종 목적지에서 이 파라미터만큼 떨어진 위치부터 stop case로 스위칭.
    # stop case부터 최종 목적지까지의 거리에 대한 pid제어가 진행. 200m 이하로 설정 권장. (뒷바퀴 중심 위치 기준)
  switch_dash_initiation_tolerance: 0.5 # stop case로 스위칭되는 위치로 인식되는 오차범위
  brake_initiation_distance: 35.0 # 최종 목적지에서 이 파라미터만큼 떨어진 위치부터 brake case로 스위칭 (뒷바퀴 중심 위치 기준)
  # 100 → 0km/h 감속 거리 : 42.68m (reference: https://help-morai-sim-standard-4-6-kr.scrollhelp.site/mssmkv/Vehicle-Specification.127011819.html#VehicleSpecification-Hyundai-IONIQ
  switch_brake_initiation_tolerance: 10 # brake case로 스위칭되는 위치로 인식되는 오차범위
  wheel_base: 2.7 # 앞바퀴와 뒷바퀴 사이의 거리(손대면 안됨)
  stanley_gain: 1.5 # stanley control gain(크면 핸들이 확확 꺾임)
  target_velocity: 60 # km/h, 목표 속도
  window_size: 25 # 속도 계획을 짤 때 몇 개의 point를 가지고 곡률을 계산할 것인가 def:25
  max_box_size: 30 # 속도 계획을 짜기 위한 곡률을 구할 때 box의 최대 크기
  road_friction: 0.4 # 도로 마찰 계수, 높을수록 곡선에서도 높은 속도를 냄
  rate: 100 # Hz, stanley control 주기

pid_control:
  # 속도 오차를 줄이기 위한 pid 제어 (driving case에서만 제어)
  velocity:
    kp: 0.3
    ki: 0.0
    kd: 0.03
  # 위치 오차를 줄이기 위한 pid 제어 (stop case에서만 제어)
  position:
    kp: 0.7
    ki: 0.0
    kd: 1.0

local_path:
  size: 100 # global path의 몇 개의 point를 가지고 local path를 생성할 것인가
  threshold: 10 # 현재 위치와 인덱스의 차이가 threshold보다 큰 포인트는 무시 (교차로에서 혼란 방지)
  rate: 100 # Hz, stanley control 주기

global_path:
  intermediate_point_distance: 3.0 # global path 간격이 클 때 몇 m 간격으로 포인트를 채울 것인가
  rate: 1 # Hz, stanley control 주기
